Objects/complexobject.c

```diff
@@ -955,13 +954,6 @@ actual_complex_new(PyTypeObject *type, PyObject *args, PyObject *kwargs)
     else if (PyErr_Occurred()) {
         return NULL;
     }
-    else if (PyComplex_Check(arg)) {
```

Comment:
This is not needed because of the tmp treatment above right? (just confirming)

Reply:
This might be accessible only for broken subclasses of the complex.

See e.g. https://github.com/python/cpython/pull/112637 on how this could be broken.  I'm not sure if we should keep such workarounds.

Reply:
> This might be accessible only for broken subclasses of the complex.

By removing this path, does broken subclasses of complex lead to a crash, or are they just plainly ignored and raise a TypeError? if so, I'm ok with removing the path.

Reply:
> does broken subclasses of complex lead to a crash

No!  Why do you think so?!

Reply:
Then let's remove it! (sorry, I was too sleepy at this time, both when I reviewed and when answered :'))